perl: Work around recompile failures

Apparently, recompiling/relinking fails under some circumstances. This
patch has been reported to work around that issue.

--- a/Makefile.SH
+++ b/Makefile.SH
@@ -867,7 +867,7 @@ $(LIBPERL): $& $(perllib_objs) $(DYNALOA
 	true)
 		$spitshell >>$Makefile <<'!NO!SUBS!'
 	rm -f $@
-	$(LD) -o $@ $(SHRPLDFLAGS) $(perllib_objs) $(DYNALOADER) $(libs)
+	$(CC) -o $@ $(SHRPLDFLAGS) $(perllib_objs) $(DYNALOADER) $(libs)
 !NO!SUBS!
 		case "$osname" in
 		aix)
